pyccl/base/deprecations.py

```diff
@@ -25,6 +26,27 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
+def mass_def_api(func):
+    """Preserve ``mass_def`` as a final argument of the decorated function."""
+    # TODO: CCLv3 - remove `mass_def=None` default from some HaloProfiles.
```

Comment:
Only some or all?

Reply:
some, because the ones who accept a concentration don't necessarily need mass_def